shivyc/tree/expr_nodes.py

```diff
@@ -424,6 +444,32 @@ def _nonarith(self, left, right, il_code):
         raise CompilerError(err, self.op.r)
 
 
+class _BitShift(_IntBinOp):
+    """Represents a `<<` and `>>` bitwise shift operators.
+    Each of operands must have integer type.
+    """
+
+    def __init__(self, left, right, op):
+        """Initialize node."""
+        super().__init__(left, right, op)
+
+    def _nonarith(self, left, right, il_code):
+        err = "invalid operand types for bitwise shift"
```

Comment:
Could you add a test that raises this error? Perhaps a line like `int* a; a << 3;` in the `error_pointer_math.c` test.